soroban-env-host/src/host/data_helper.rs

```diff
@@ -236,18 +236,23 @@ impl Host {
     pub(crate) fn extend_contract_instance_and_code_ttl_from_contract_id(
         &self,
         contract_id: &Hash,
-        threshold: u32,
-        extend_to: u32,
+        instance_threshold: u32,
+        code_threshold: Option<u32>,
+        extend_instance_to: u32,
+        extend_code_to: Option<u32>,
```

Comment:
I think in general having optional fields in function signatures are confusing. 
I would prefer the other way around: make these inputs non-optional and change the existing callers to explicitly passing them in. The change surface area should be similar and it's less confusing.

Reply:
Yeah if we go with https://github.com/stellar/stellar-protocol/discussions/1447#discussioncomment-8591665 this fn would be "duplicated" into `extend_contract_code_ttl_from_contract_id` for bumping contract code and the existing helper modified to `extend_contract_instance_ttl_from_contract_id` which only bumps the instance.